aws/resource_aws_securityhub_standards_control.go

```diff
@@ -0,0 +1,140 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"path"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/securityhub"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAwsSecurityHubStandardsControl() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAwsSecurityHubStandardsControlCreate,
+		ReadContext:   resourceAwsSecurityHubStandardsControlRead,
+		UpdateContext: resourceAwsSecurityHubStandardsControlUpdate,
+		DeleteContext: resourceAwsSecurityHubStandardsControlDelete,
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+			"control_status": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					securityhub.ControlStatusEnabled,
+					securityhub.ControlStatusDisabled,
+				}, false),
+			},
+			"disabled_reason": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"control_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"control_status_updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"related_requirements": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"remediation_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"severity_rating": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"title": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsSecurityHubStandardsControlCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	d.SetId(d.Get("arn").(string))
+
+	resourceAwsSecurityHubStandardsControlUpdate(ctx, d, meta)
+	return resourceAwsSecurityHubStandardsControlRead(ctx, d, meta)
+}
+
+func resourceAwsSecurityHubStandardsControlRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).securityhubconn
+
+	arn := d.Get("arn").(string)
+	subscription_arn := path.Dir(strings.ReplaceAll(arn, "control", "subscription"))
```

Comment:
Maybe use `arn.Parse`, string replacement and then `arn.String()`?

Reply:
ARN `Resource` field is too wide to make it simple – `control/cis-aws-foundations-benchmark/v/1.2.0/1.12`. I don't think it brings more clarity, TBH.